# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../pump/off-by-one.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- pump-0.8.24/pump.c.orig	2005-10-11 19:35:59.000000000 +0200
+++ pump-0.8.24/pump.c	2008-08-25 14:42:48.000000000 +0200
@@ -598,7 +598,7 @@
 		    if (intf[i].set & PUMP_NETINFO_HAS_HOSTNAME)
 			strncpy(cmd.u.status.hostname,
 			    intf->hostname, sizeof(cmd.u.status.hostname));
-		    cmd.u.status.hostname[sizeof(cmd.u.status.hostname)] = '\0';
+		    cmd.u.status.hostname[sizeof(cmd.u.status.hostname) - 1] = '\0';
 
 		    if (intf[i].set & PUMP_NETINFO_HAS_DOMAIN)
 			strncpy(cmd.u.status.domain,
